lib/ai_bot/personas/persona.rb

```diff
@@ -122,16 +122,24 @@ def find_tool(partial)
           tool_klass = available_tools.find { |c| c.signature.dig(:name) == function_name }
           return false if tool_klass.nil?
 
-          arguments =
-            tool_klass.signature[:parameters]
-              .to_a
-              .reduce({}) do |memo, p|
-                argument = parsed_function.at(p[:name])&.text
-                next(memo) unless argument
-
-                memo[p[:name].to_sym] = argument
-                memo
-              end
+          arguments = {}
+          tool_klass.signature[:parameters].to_a.each do |param|
+            name = param[:name]
+            value = parsed_function.at(name)&.text
+
+            if param[:type] == "array" && value
+              value =
+                begin
+                  JSON.parse(value)
+                rescue JSON::ParserError
+                  nil
+                end
+            end
+
+            if value
+              arguments[name.to_sym] = value if value
+            end
```

Comment:
There is a double `if value` conditional here. Not sure why Github doesn't allow me to use code suggestion here.

Reply:
Oops will fix